src/main/scala/shine/C/AST/Printer.scala

```diff
@@ -274,136 +290,137 @@ class CPrinter extends Printer {
-    print("((")
+    print("(")
     if (al.t.const) { print("const ") }
     print(s"${typeName(al.t.getBaseType)}[${ al.t.getSizes match {
       case None => ""
-      case Some(s) => toString(s)
+      case Some(s) => printArithExpr(s, parenthesize = false)
     } }]")
     print("){")
     var first = true
     al.inits.foreach { e =>
       if (first) { first = false  }
       else { print(", ") }
-      printExpr(e)
+      printExpr(e, parenthesize = false)
     }
-    print("})")
+    print("}")
   }
 
   private def printRecordLiteral(rl: RecordLiteral): Unit = {
     print(s"(${typeName(rl.t)})")
     print("{ ")
-    printExpr(rl.fst)
+    printExpr(rl.fst, parenthesize = false)
     print(", ")
-    printExpr(rl.snd)
+    printExpr(rl.snd, parenthesize = false)
     print(" }")
   }
 
-  private def printArithmeticExpr(a: ArithmeticExpr): Unit = {
-    print(toString(a.ae))
+  private def printArithmeticExpr(a: ArithmeticExpr, parenthesize: Boolean): Unit = {
+    print(printArithExpr(a.ae, parenthesize))
   }
 
   // Types
   def typeName(t: Type): String = t.toString
 
-  override def toString(e: ArithExpr) : String = {
+  override def printArithExpr(e: ArithExpr, parenthesize: Boolean) : String = {
     e match {
       case Cst(c) => c.toString
       case Pow(b, ex) =>
         ex match {
-          case Cst(2) => s"(${toString(b)} * ${toString(b)})"
-          case _ => "(int)pow((float)" + toString (b) + ", " + toString (ex) + ")"
+          case Cst(2) =>
+            (if (parenthesize) "(" else "") +
+              s"${printArithExpr(b, parenthesize = true)} * ${printArithExpr(b, parenthesize = true)}" +
+              (if (parenthesize) ")" else "")
+          case _ =>
+            s"(int)pow((float) ${printArithExpr(b, parenthesize = false)}, " +
+              s"${printArithExpr(ex, parenthesize = false)})"
         }
-      case Log(b, x) => "(int)log"+b+"((float)"+toString(x)+")"
-      case Prod(es) => "(" + es.foldLeft("1")( (s: String, e: ArithExpr) => {
+      case Log(b, x) => s"(int)log$b((float)${printArithExpr(x, parenthesize = false)})"
+      case Prod(es) => (if (parenthesize) "(" else "") + es.foldLeft("1")((s, e) => {
         s + (e match {
-          case Pow(b, Cst(-1)) => " / (" + toString(b) + ")"
-          case _ => " * " + toString(e)
+          case Pow(b, Cst(-1)) => " / " + printArithExpr(b, parenthesize = true)
+          case _ => " * " + printArithExpr(e, parenthesize = true)
         })
-      } ).drop(4) + ")" // drop(4) removes the initial "1 * "
-      case Sum(es) => "(" + es.map(toString).reduce( _ + " + " + _  ) + ")"
-      case Mod(a,n) => "(" + toString(a) + " % " + toString(n) + ")"
+      } ).drop(4) + (if (parenthesize) ")" else "") // drop(4) removes the initial "1 * "
+      case Sum(es) =>
+        (if (parenthesize) "(" else "") +
+          s"${es.map(printArithExpr(_, parenthesize = true)).reduce(_+" + "+_)}" +
+          (if (parenthesize) ")" else "")
```

Comment:
Maybe a helper can be defined to avoid repeating this many times:
```scala
def maybeParenthesize(s: String): String =
  (if (parenthesize) "(" else "") + s + (if (parenthesize) ")" else "")
```

Reply:
I have addressed this by introducing helper functions.

Reply:
@Bastacyclop maybe you can have a brief look and if you are happy approve the PR.